src/schema/types/user.ts

```diff
@@ -57,10 +57,20 @@ export const UserType = new GraphQLObjectType({
         applications: {
             type: new GraphQLList(ApplicationType),
             async resolve(parent, args, context) {
+                let isAdmin: boolean = false;
                 const ability: AppAbility = context.user.ability;
                 const roles = await Role.find().where('_id').in(parent.roles);
-                const applications = roles.map(x => mongoose.Types.ObjectId(x.application));
-                return Application.accessibleBy(ability, 'read').where('_id').in(applications);
+                roles.map(x => {
+                    if  (x.title === 'admin') {
```

Comment:
@Hugo-Plissonneau the name of an admin role doesn't need to be 'admin', we can also define other roles that would be considered as admin role for applications